### PR TITLE
feat: APIs for the new campus card system

### PR DESCRIPTION
**Pull request by 84634E1A607A**

### CODE REVIEW

File: src/lib/card.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    const redirectUrl = await roam(helper, "card", "eea30cbedcaf97c69d28b2d92f22a259");
+    const ticket = new RegExp(/ticket=(\w*?)$/).exec(redirectUrl)![1];
+    const token = (await fetchWithParse(CARD_LOGIN_URL, {ticket: ticket})).token;
+    setCookie("token", token);
+    accountBaseInfo.user = parseResultData(await uFetch(CARD_USER_BY_TOKEN_URL)).loginuser;
+    return token;
+};
+
+export const cardGetInfo = async (helper: InfoHelper): Promise<CardInfo> => {
+    if (accountBaseInfo.user === "") {
```
Comment (UNIDY2002) [inline src/lib/card.ts:43]: It is not a good practice to test the login status in this way, as the token may expire and the APP may live longer than the token.

**Review by UNIDY2002 — CHANGES_REQUESTED** [review]

File: src/lib/card.ts (hunk 1 above)
Comment (84634E1A607A) [reply src/lib/card.ts:43]: Fine. Will change that later.

**Review by UNIDY2002 — COMMENTED** [review]
Please also update the API list in `README.md` :)

**Review by UNIDY2002 — APPROVED** [review]
LGTM

**Pull request merged**